cmd/cli/mesh_list.go

```diff
@@ -64,13 +74,26 @@ func (l *meshListCmd) run() error {
 
 	w := newTabWriter(l.out)
 
-	fmt.Fprintln(w, "\nMESH NAME\tNAMESPACE\tCONTROLLER PODS\tVERSION")
+	fmt.Fprintln(w, "\nMESH NAME\tNAMESPACE\tCONTROLLER PODS\tVERSION\tSMI SUPPORTED")
+
 	for _, elem := range list.Items {
 		m := elem.ObjectMeta.Labels["meshName"]
 		ns := elem.ObjectMeta.Namespace
 		x := getNamespacePods(l.clientSet, m, ns)
+
+		var smiList []string
+		smiMap, err := getSupportedSmiForControllerPod(x["Pods"][0], ns, l)
```

Comment:
@shashankram @draychev For now, we can assume that each mesh only has one controller pod right?

`x["Pods"][0]` => Gets the supported SMI by querying the first controller pod. That would be valid for now, but what would happen if in the future if there are multiple controller pods for each mesh? This line of code may cause issues then.

Reply:
We should not introduce such an assumption because there can be multiple instances of OSM controller within the same mesh for scalability/HA reasons. The `replicaCount` for the controller can be changed in the `values.yaml` file for this purpose.

I don't think multiple controllers should have any impact on what this command displays though, because all of them will be using the same controller image hence supporting the same SMI version. In the future, when we do have to support hitless controller upgrades, there might be a situation where controllers belong to different versions (yet to be upgraded), but that would be for a small amount of time and we can disregard that for the moment.

Reply:
You could also go the extra mile, and list the smi supported for all OSM instances running in the mesh.... :)

Reply:
@shashankram and @eduser25 I've added checks for the `"Pods"` key and length checks. I agree with @shashankram in that the number of replicas for the osm controller won't really affect the output of this command, except for a tiny window during osm upgrade. I agree with @shashankram in that we can disregard this small window for the moment, plus since I've added those checks already, the code should be resilient enough.

![image](https://user-images.githubusercontent.com/13926417/116328502-0239ea80-a77e-11eb-97a2-47b108078957.png)

Reply:
@eduser25 listing the smi versions for each controller pod would clutter the output of this command (too verbose). Also as stated above, I think this implementation is good enough since versions would only diverge for a tiny window.